b2

```diff
@@ -763,10 +763,75 @@ class AuthInfoCache(AbstractCache):
         self.info.refresh_entire_bucket_name_cache(self._name_id_iterator(buckets))
 
 
+## B2RawApi
+
+
+class B2RawApi(object):
+
+    """
+    Provides access to the B2 web APIs, exactly as they are provided by B2.
+
+    Requires that you provide all necessary URLs and auth tokens for each call.
+
+    Each API call decodes the returned JSON and returns a dict.
+
+    For details on what each method does, see the B2 docs:
+        https://www.backblaze.com/b2/docs/
+
+    This class is intended to be a super-simple, very thin layer on top
+    of the HTTP calls.  It can be mocked-out for testing higher layers.
+    And this class can be tested by exercising each call just once,
+    which is relatively quick.
+    """
+
+    def authorize_account(self, realm_url, account_id, application_key):
+        auth = 'Basic '+ base64.b64encode('%s:%s' % (account_id, application_key))
+        url = self._build_url(realm_url, 'b2_authorize_account')
+        return post_json(url, {}, auth)
+
+    def create_bucket(self, api_url, account_id, account_auth_token, bucket_name, bucket_type):
+        url = self._build_url(api_url, 'b2_create_bucket')
+        params = {
+            'accountId' : account_id,
+            'bucketName' : bucket_name,
+            'bucketType' : bucket_type,
+            }
+        return post_json(url, params, account_auth_token)
+
+    # TODO: move the rest of the calls from B2Api
+
+    def _build_url(self, url_base, api_name):
+        return url_base + '/b2api/v1/' + api_name
+
 ## B2Api
 
 class B2Api(object):
+
+    """
+    Provides high-level access to the B2 API.
+
+    Adds an object-oriented layer on top of the raw API, so that
+    buckets and files returned are Python objects with accessor
+    methods.
+
+    Also,  keeps a cache of information needed to access the service,
+    such as auth tokens and upload URLs.
+    """
+
+    # TODO: move HTTP code out to B2RawApi
+    # TODO: ConsoleTool passes the account info cache into the constructor
+    # TODO: provide method to get the account info cache (so ConsoleTool can save it)
```

Comment:
Indeed. They can pass `account_info` and `cache` arguments to `B2Api` for exactly that purpose. `InMemoryCache` is implemented already, but I haven't gotten around to `InMemoryAccountInfo` yet.